pkg/chain/chain.go

```diff
@@ -1,27 +1,83 @@
-// Package eth contains interface for interaction with an ethereum blockchain
-// along with structures reflecting events emitted on an ethereum blockchain.
-package eth // TODO: rename; this can be any host chain
+// Package chain contains interface for interaction with a host chain along with
+// structures reflecting events emitted on such a chain.
+package chain // TODO: rename; this can be any host chain
 
 import (
+	cecdsa "crypto/ecdsa"
+	"fmt"
 	"math/big"
 	"time"
 
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/keep-network/keep-common/pkg/subscription"
 	"github.com/keep-network/keep-core/pkg/chain"
 	"github.com/keep-network/keep-ecdsa/pkg/ecdsa"
 )
 
-// Handle represents a handle to an ethereum blockchain.
+type KeepApplicationID interface {
+	fmt.Stringer
+}
+
+type KeepMemberID interface {
+	fmt.Stringer
+
+	// KeepMemberIDs should be convertible to their associated OperatorID. This
+	// may be a 1-to-1 correspondence, i.e. a simple interface cast, or it may
+	// be an internally-tracked association on the host chain.
+	OperatorID() OperatorID
+}
+
+type OperatorID interface {
+	fmt.Stringer
+
+	// OperatorIDs should be convertible to an associated KeepMemberID given the
+	// id of the keep in question. This may be a 1-to-1 correspondence, i.e. a
+	// simple interface cast, or it may be an internally-tracked association on
+	// the host chain. The conversion _may but is not guaranteed to be_ nil if
+	// the operator is not in the specified keep.
+	KeepMemberID(keepID KeepID) KeepMemberID
+}
+
+// Should be chain-specific, unique across chains, and simple ([A-Za-z0-9-_.]).
+// This can generally mean prefixing the common name of the chain followed by a
+// dash followed by the chain-specific identifier (e.g., the contract address
+// for Ethereum). For legacy reasons, Ethereum KeepIDs should always be the bare
+// Keep contract address.
+//
+// Should not be expected to be comparable between types or amongst themselves.
```

Comment:
Should not be expected to be comparable because the bare Ethereum-format address on Celo chain obtained from `fmt.Stringer` may look the same as the bare Ethereum-format address on Ethereum chain obtained from `fmt.Stringer` but they are IDs of two separate keeps?

Reply:
Yes, and meaning to say that `==` should not be expected to work “correctly”. Right now as these are structured, the only way folks should probably compare them is using `.String() == .String()`, or by casting internally to known internal types (which of course risks problems if someone manages to cross an id from one chain to another).